### PR TITLE
shanoir-issue#2446: add list of studies of subject on subjectDelete

### PR DESCRIPTION
**Pull request by pierrehenri-dauvergne**
How to test:

Open subject-list
Click on the delete button
The popup should display the studies in which the subject belong:
image

Do the same but not from the subject-list but from a view with a tree:
image

### CODE REVIEW

**Comment by jcomedouteau** [comment]
Hey, it works from subject list, but not from subject details:
image

File: shanoir-ng-front/src/app/subjects/subject-list/subject-list.component.ts [hunk 1]
```diff
@@ -90,4 +90,13 @@ export class SubjectListComponent extends EntityListComponent<Subject> {
             subject.subjectStudyList.filter(ss => this.studiesICanAdmin?.includes(ss.study.id)).length > 0
         );
     }
+
+    getOnDeleteConfirmMessage(entity: Subject): Promise<string> {
+        let studyListStr : string = "\n\nThis subject belong in studies: \n- ";
```
Comment (michaelkain) [inline shanoir-ng-front/src/app/subjects/subject-list/subject-list.component.ts:95]: same here: please change to "This subject belongs to the studies: "

File: shanoir-ng-front/src/app/shared/components/entity/entity.abstract.service.ts [hunk 2]
```diff
@@ -71,40 +70,43 @@ export abstract class EntityService<T extends Entity> implements OnDestroy {
             .toPromise();
     }
 
-    deleteWithConfirmDialog(name: string, entity: Entity): Promise<boolean> {
+    deleteWithConfirmDialog(name: string, entity: Entity, studyListStr?: string): Promise<boolean> {
+        let dialogTitle : string = 'Delete ' + name;
+        let dialogMsg : string = 'Are you sure you want to delete the ' + name
```
Comment (michaelkain) [inline shanoir-ng-front/src/app/shared/components/entity/entity.abstract.service.ts:75]: @pierrehenri-dauvergne please change: "Are you sure you want to finally delete the "

File: shanoir-ng-front/src/app/shared/components/entity/entity-list.component.abstract.ts [hunk 3]
```diff
@@ -115,12 +118,22 @@ export abstract class EntityListComponent<T extends Entity> implements OnDestroy
...
 
     protected openDeleteConfirmDialog = (entity: T) => {
-        this.confirmDialogService
-            .confirm(
-                'Delete ' + this.ROUTING_NAME,
-                'Are you sure you want to delete the ' + this.ROUTING_NAME
-                + (entity['name'] ? ' "' + entity['name'] + '"' : ' with id n° ' + entity.id) + ' ?'
-            ).then(res => {
+        let dialogTitle : string = 'Delete ' + this.ROUTING_NAME;
+        let dialogMsg : string = 'Are you sure you want to delete the ' + this.ROUTING_NAME
```
Comment (michaelkain) [inline shanoir-ng-front/src/app/shared/components/entity/entity-list.component.abstract.ts:122]: @pierrehenri-dauvergne please change: "Are you sure you want to finally delete the "

File: shanoir-ng-front/src/app/subjects/subject-list/subject-list.component.ts [hunk 4]
```diff
@@ -90,4 +90,13 @@ export class SubjectListComponent extends EntityListComponent<Subject> {
             subject.subjectStudyList.filter(ss => this.studiesICanAdmin?.includes(ss.study.id)).length > 0
         );
     }
+
+    getOnDeleteConfirmMessage(entity: Subject): Promise<string> {
+        let studyListStr : string = "\n\nThis subject belongs to the studies:: \n- ";
```
Comment (michaelkain) [inline shanoir-ng-front/src/app/subjects/subject-list/subject-list.component.ts:95]: Hi @pierrehenri-dauvergne : two "::" :-)?

**Comment by pierrehenri-dauvergne** [comment]
Also add a warning text that says "Deleting a subject will delete all its content (examination, acquisition, dataset)."

File: shanoir-ng-front/src/app/shared/components/entity/entity-list.component.abstract.ts [hunk 5]
```diff
@@ -116,12 +119,22 @@ export abstract class EntityListComponent<T extends Entity> implements OnDestroy
...
-        this.confirmDialogService
-            .confirm(
-                'Delete ' + this.ROUTING_NAME,
-                'Are you sure you want to delete the ' + this.ROUTING_NAME
-                + (entity['name'] ? ' "' + entity['name'] + '"' : ' with id n° ' + entity.id) + ' ?'
-            ).then(res => {
+        let dialogTitle : string = 'Delete ' + this.ROUTING_NAME;
+        let dialogMsg : string = 'Are you sure you want to finally delete the ' + this.ROUTING_NAME
+            + (entity['name'] ? ' \"' + entity['name'] + '\"' : ' with id n° ' + entity.id) + ' ?';
+
```
Comment (michaelkain) [inline shanoir-ng-front/src/app/shared/components/entity/entity-list.component.abstract.ts:125]: Hi @pierrehenri-dauvergne: please add here, after the list of studies:
"Attention: this action deletes all datasets from ALL studies listed above."

File: shanoir-ng-front/src/app/shared/components/entity/entity.abstract.service.ts [hunk 6]
```diff
@@ -71,40 +70,43 @@ export abstract class EntityService<T extends Entity> implements OnDestroy {
             .toPromise();
     }
 
-    deleteWithConfirmDialog(name: string, entity: Entity): Promise<boolean> {
+    deleteWithConfirmDialog(name: string, entity: Entity, studyListStr?: string): Promise<boolean> {
+        let dialogTitle : string = 'Delete ' + name;
+        let dialogMsg : string = 'Are you sure you want to finally delete the ' + name
+            + (entity['name'] ? ' \"' + entity['name'] + '\"' : ' with id n° ' + entity.id) + ' ?';
+
```
Comment (michaelkain) [inline shanoir-ng-front/src/app/shared/components/entity/entity.abstract.service.ts:77]: Hi @pierrehenri-dauvergne: please add here, after the list of studies:
"Attention: this action deletes all datasets from ALL studies listed above."

**Comment by michaelkain** [comment]
Hi @pierrehenri-dauvergne, do you think you might solve the conflict please? Thank you, Michael

**Review by michaelkain — APPROVED** [review]
Approved by code review and local tests, that all worked.

**Pull request merged**